src/sentry/utils/dates.py

```diff
@@ -102,13 +104,39 @@ def parse_stats_period(period):
     )
 
 
-def get_rollup_from_request(request, params, default_interval, error, top_events=0):
+def get_interval_from_range(date_range: timedelta, high_fidelity: bool) -> str:
+    if date_range >= timedelta(days=60):
+        return "4h" if high_fidelity else "1d"
+
+    if date_range >= timedelta(days=30):
+        return "1h" if high_fidelity else "4h"
+
+    if date_range > timedelta(days=1):
+        return "30m" if high_fidelity else "1h"
+
+    if date_range > timedelta(hours=1):
+        return "1m" if high_fidelity else "5m"
+
+    return "5m" if high_fidelity else "15m"
+
+
+def get_rollup_from_request(
+    request: HttpRequest,
+    params: Mapping[str, Any],
+    default_interval: Union[None, str],
+    error: Exception,
+    top_events: int = 0,
+) -> int:
+    date_range = params["end"] - params["start"]
+
+    if default_interval is None:
+        default_interval = get_interval_from_range(date_range, False)
```

Comment:
Looks like the trends endpoint tests is not a fan of this change.

Reply:
Fixed by adding the interval parameter explicitly. I think this is OK